src/exporters/s5.py

```diff
@@ -0,0 +1,488 @@
+    #             return '{:02d}'.format(value)
+    #         elif key == 'time':
+    #             return '{:02d}:00'.format(value)
+    #     return str(value)
+
+    def get_pressure_levels(
+        self, pressure_levels: Optional[List[int]] = None
+    ) -> Optional[Dict]:
+        if self.pressure_level:
+            # make sure the dataset requires pressure_levels
+            assert "pressure_level" in [
+                k for k in self.dataset_reference.keys()
+            ], f"\
+            {self.dataset} has no 'pressure_level' keys. Cannot assign pressure levels"
+
+            # make sure the pressure levels are legitimate choices
+            assert all(
+                np.isin(pressure_levels, self.dataset_reference["pressure_level"])
+            ), f"\
+            {pressure_levels} contains invalid pressure levels!\
+            Must be one of:{self.dataset_reference['pressure_level']}"
+
+            return {"pressure_level": [str(pl) for pl in pressure_levels]}
+        else:
+            return None
+
+    def get_valid_variables(self) -> List:
+        """ get `valid_variables` for this S5Exporter object """
+        # print(self.dataset_reference['variable'])
+        return self.dataset_reference["variable"]
+
+    def create_selection_request(
+        self,
+        variable: str,
+        max_leadtime: int,
+        min_year: int = 1993,
+        max_year: int = 2019,
+        min_month: int = 1,
+        max_month: int = 12,
+        selection_request: Optional[Dict] = None,
+    ) -> Dict:
+        """Build up the selection_request dictionary with defaults
+
+        Returns
+        ----------
+        selection_dict: dict
+            A dictionary with all the time-related arguments of the
+            selection dict filled out
+
+        Note:
+        ----
+        - This is a FULL version of the dictionary that is returned. Will likely
+         lead to errors if you don't chunk the data (e.g. by month) before
+         sending the requests to cdsapi
+        - Some attributes are used to create the default arguments
+            (self.product_type, self.granularity, self.dataset, self.pressure_level)
+        """
+        assert (
+            variable in self.dataset_reference["variable"]
+        ), f"\
+            Variable: {variable} is not in the valid variables for this \
+            dataset. Valid variables: {self.dataset_reference['variable']}"
+
+        # setup the default selection request
+        processed_selection_request = {
+            "format": "grib",
+            "originating_centre": "ecmwf",
+            "system": "5",
+            "variable": [variable],
+        }
+
+        # add product_type if required
+        if self.product_type is not None:
+            processed_selection_request.update({"product_type": [self.product_type]})
+
+        # get the initialisation time information
+        init_times_dict = self.get_s5_initialisation_times(
```

Comment:
???????? @gabrieltseng 

How do I specify that these are `Optional` without mypy then complaining that I am comparing `None` type to 
```
src/exporters/s5.py:405: error: Argument 2 to "get_s5_initialisation_times" of "S5Exporter" has incompatible type "Optional[int]"; expected "int"
src/exporters/s5.py:405: error: Argument 3 to "get_s5_initialisation_times" of "S5Exporter" has incompatible type "Optional[int]"; expected "int"
src/exporters/s5.py:405: error: Argument 4 to "get_s5_initialisation_times" of "S5Exporter" has incompatible type "Optional[int]"; expected "int"
src/exporters/s5.py:405: error: Argument 5 to "get_s5_initialisation_times" of "S5Exporter" has incompatible type "Optional[int]"; expected "int"
```

If I change to `Optional[int] = ...` for all the arguments in `get_s5_initialisation_times`
```
src/exporters/s5.py:220: error: Unsupported operand types for <= ("int" and "None")
src/exporters/s5.py:220: note: Left operand is of type "Optional[int]"
src/exporters/s5.py:224: error: Unsupported operand types for >= ("int" and "None")
src/exporters/s5.py:224: note: Left operand is of type "Optional[int]"
src/exporters/s5.py:229: error: Argument 1 to "range" has incompatible type "Optional[int]"; expected "int"
src/exporters/s5.py:229: error: Unsupported operand types for + ("None" and "int")
src/exporters/s5.py:229: note: Left operand is of type "Optional[int]"
src/exporters/s5.py:230: error: Argument 1 to "range" has incompatible type "Optional[int]"; expected "int"
src/exporters/s5.py:230: error: Unsupported operand types for + ("None" and "int")
src/exporters/s5.py:230: note: Left operand is of type "Optional[int]"
```

Reply:
So the trouble was that the default gets overridden by the None argument being passed to `get_s5_initialization_times`. The solution was to not pass the argument it if was None.